squad/api/rest.py

```diff
@@ -799,16 +811,23 @@ def get_url(self, *args):
         testrun = args[0]
         statuses = testrun.status.all()
         if len(statuses) > 0:
-            tr_status = testrun.status.all()[0]
-            num_tests = (
-                tr_status.tests_pass + tr_status.tests_fail + tr_status.tests_skip + tr_status.tests_xfail
-            )
-            if num_tests > 0:
-                return super().get_url(*args)
+            for s in statuses:
```

Comment:
I think it's fine querying only the first `status` like it used to be. Then I guess it makes sense to do:
```python
queryset = TestRun.objects.prefetch_related(	
        Prefetch("status", queryset=Status.objects.filter(suite=None))	
).order_by("-id")
```
instead of the new prefetch you implemented:

```python
queryset = TestRun.objects.prefetch_related("status").order_by("-id")
```

One less loop in the code :)